src/Cli/dotnet/commands/dotnet-workload/install/NetSdkManagedInstaller.cs

```diff
@@ -269,6 +270,13 @@ public void GarbageCollectInstalledWorkloadPacks()
                         unneededBandRecords = unneededBandRecords.Append(currentBandRecordPath);
                     }
 
+                    if (!unneededBandRecords.Any())
+                    {
+                        continue;
+                    }
+
+                    // Save the pack info in case we need to delete the pack
```

Comment:
Any test coverage for adding pack in for to installation record? This is to cover manifest pack info change in between versions?

Reply:
Yes, good idea, I'll add a test.